astropy/cosmology/flrw/base.py

```diff
@@ -87,25 +106,25 @@ def scale_factor0(self):
         the redshift of interest for that paper. Explicitly using :math:`a_0` in both
         calculation and code avoids ambiguity.
         """
-        return u.Quantity(self.scale_factor(0), unit=u.one)
+        return self.scale_factor(0) << u.one
 
     @deprecated_keywords("z", since="7.0")
-    def scale_factor(self, z):
+    def scale_factor(self, z) -> Quantity | NDArray[Any] | float:
         """Scale factor at redshift ``z``.
 
         The scale factor is defined as :math:`a = 1 / (1 + z)`.
 
         Parameters
         ----------
-        z : Quantity-like ['redshift'], array-like
+        z : Quantity-like ['redshift'], array-like, positional-only
             Input redshift.
 
             .. versionchanged:: 7.0
                 Passing z as a keyword argument is deprecated.
 
         Returns
         -------
-        a : ndarray or float
+        ndarray or float
```

Comment:
this doesn't seem consistent with the added annotation (no `Quantity` ?)

Reply:
✅